tests/lua/metatable.lua

```diff
@@ -164,24 +164,44 @@ test.shared_table_with_metatable.as_shared_table = function()
 
     -- Only __index metamethod
     mt.__index = function(t, key)
-        return "mt_" .. effil.rawget(t, key)
+        return "mt_" .. effil.rawget(t, key .. "_origin")
     end
-    share.table_key = "table_value"
+    share.table_key_origin = "table_value"
     test.equal(share.table_key, "mt_table_value")
 
     -- Both __index and __newindex metamethods
     mt.__newindex = function(t, key, value)
         effil.rawset(t, key, "mt_" .. value)
     end
     share.table_key = "table_value"
-    test.equal(share.table_key, "mt_mt_table_value")
+    test.equal(share.table_key, "mt_table_value")
 
     -- Remove __index, use only __newindex metamethods
     mt.__index = nil
     share.table_key = "table_value"
     test.equal(share.table_key, "mt_table_value")
 end
 
+test.shared_table_with_metatable.table_as_index = function()
```

Comment:
There is no test with `function` as `__index`.

Reply:
All previously existing tests here are for `function`.